railties/test/application/configuration_test.rb

```diff
@@ -1304,6 +1304,32 @@ def index
       assert_equal false, ActionView::Resolver.caching?
     end
 
+    test "ActionController::Base.allow_open_redirects is false by default for new apps" do
+      app "development"
+
+      assert_equal false, ActionController::Base.allow_open_redirects
+    end
+
+    test "ActionController::Base.allow_open_redirects is true by default for upgraded apps" do
+      remove_from_config '.*config\.load_defaults.*\n'
+      add_to_config 'config.load_defaults "6.1"'
+      app "development"
+
+      assert_equal true, ActionController::Base.allow_open_redirects
+    end
+
+    test "ActionController::Base.allow_open_redirects can be configured in the new framework defaults" do
+      remove_from_config '.*config\.load_defaults.*\n'
+
+      app_file "config/initializers/new_framework_defaults_6_2.rb", <<-RUBY
+        Rails.application.config.action_controller.allow_open_redirects = false
+      RUBY
+
+      app "development"
+
+      assert_equal false, ActionController::Base.allow_open_redirects
```

Comment:
The fact that `false` is when it's on is confusing, so let's flip this default to be `raise_on_open_redirects = true`.